src/app/app.component.ts

```diff
@@ -1,20 +1,62 @@
-import {Component, OnInit} from '@angular/core';
-import {Router} from '@angular/router';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { Link } from './app.api';
+import { NavigationService } from './navigation.service';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.css']
+  styleUrls: ['./app.component.css'],
 })
-export class AppComponent implements OnInit {
-  constructor(private readonly router: Router) {
+export class AppComponent implements OnInit, OnDestroy {
+  /**
+   * List of all valid links
+   */
+  links: Link[];
+
+  /**
+   * For easier handling of the navigation via the button the index of
+   * the currently selected index is used
+   */
+  selectedLinkIndex: number;
```

Comment:
```suggestion
  links$: Observable<Link[]>;

  /**
   * For easier handling of the navigation via the button the index of
   * the currently selected index is used
   */
  selectedLinkIndex = 0;
```
Ich würde direkt initialisieren und dafür die subscription sparen. Somit kein onDestroy und kein `links` array notwendig, stattdessen, siehe nächster Kommentar.